torch/_dynamo/variables/torch.py

```diff
@@ -416,6 +422,27 @@ def handle_tracing_state_functions(
                 tx.mark_inconsistent_side_effects()
             return ConstantVariable.create(tracing_state_functions[self.value])
 
+        @register(*dispatch_key_set_functions)
+        def handle_dispatch_key_set_functions(
+            self, tx: "InstructionTranslator", *args, **kwargs
+        ):
+            assert not kwargs
+            if self.value in (torch._C._dispatch_keys,):
+                assert len(args) == 1
+                assert isinstance(args[0], variables.TensorVariable)
+                fake_tensor = args[0].proxy.node.meta["example_value"]
+                dks = self.value(fake_tensor)
+                # Remove Python and PythonTLSSnapshot from the dispatch key set since they are from FakeTensor propogation.
+                dks = (
+                    dks
+                    - torch._C.DispatchKeySet(torch._C.DispatchKey.Python)
+                    - torch._C.DispatchKeySet(torch._C.DispatchKey.PythonTLSSnapshot)
+                )
+                return DispatchKeySetVariable.create(dks)
```

Comment:
Only remove these if `isinstance(fake_tensor, FakeTensor)`. If Tensor subclasses are around, then it's reasonable for Python to be in the keyset.